x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/DeleteShutdownNodeAction.java

```diff
@@ -32,29 +34,35 @@ public static class Request extends AcknowledgedRequest<Request> {
 
         private final String nodeId;
 
-        public Request(String nodeId) {
-            super(TRAPPY_IMPLICIT_DEFAULT_MASTER_NODE_TIMEOUT, DEFAULT_ACK_TIMEOUT);
+        public Request(TimeValue masterNodeTimeout, TimeValue ackTimeout, String nodeId) {
+            super(masterNodeTimeout, ackTimeout);
             this.nodeId = nodeId;
         }
 
-        public Request(StreamInput in) throws IOException {
-            super(TRAPPY_IMPLICIT_DEFAULT_MASTER_NODE_TIMEOUT, DEFAULT_ACK_TIMEOUT);
-            if (in.getTransportVersion().isPatchFrom(TransportVersions.V_8_13_4)
-                || in.getTransportVersion().isPatchFrom(TransportVersions.SHUTDOWN_REQUEST_TIMEOUTS_FIX_8_14)
-                || in.getTransportVersion().onOrAfter(TransportVersions.SHUTDOWN_REQUEST_TIMEOUTS_FIX)) {
-                // effectively super(in):
-                setParentTask(TaskId.readFromStream(in));
-                masterNodeTimeout(in.readTimeValue());
-                ackTimeout(in.readTimeValue());
+        @UpdateForV9 // inline when bwc no longer needed
+        public static Request readFrom(StreamInput in) throws IOException {
+            if (serializedWithParentTaskAndTimeouts(in.getTransportVersion())) {
+                return new Request(in);
+            } else {
+                return new Request(TimeValue.THIRTY_SECONDS, TimeValue.THIRTY_SECONDS, in);
             }
+        }
+
+        private Request(StreamInput in) throws IOException {
+            super(in);
+            assert serializedWithParentTaskAndTimeouts(in.getTransportVersion());
             this.nodeId = in.readString();
         }
 
+        @UpdateForV9 // remove when bwc no longer needed
+        private Request(TimeValue masterNodeTimeout, TimeValue ackTimeout, StreamInput in) throws IOException {
+            this(masterNodeTimeout, ackTimeout, in.readString());
+            assert serializedWithParentTaskAndTimeouts(in.getTransportVersion()) == false;
+        }
+
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            if (out.getTransportVersion().isPatchFrom(TransportVersions.V_8_13_4)
-                || out.getTransportVersion().isPatchFrom(TransportVersions.SHUTDOWN_REQUEST_TIMEOUTS_FIX_8_14)
-                || out.getTransportVersion().onOrAfter(TransportVersions.SHUTDOWN_REQUEST_TIMEOUTS_FIX)) {
+            if (serializedWithParentTaskAndTimeouts(out.getTransportVersion())) {
```

Comment:
Nit: the name `serializedWithParentTaskAndTimeouts` reads a bit strange when it is used in a method where serialization is yet to happen. Maybe `supportsSerializeWithParentTaskAndTimeouts`?